syntax = "proto2";

import "nanopb.proto";

message SubMessage {
    option (nanopb_msgopt).callback_function = "SubMessage_callback";

    optional string stringvalue = 1;
    repeated int32 int32value = 2;
    repeated fixed32 fixed32value = 3;
    repeated fixed64 fixed64value = 4;
}

message TestMessage {
    option (nanopb_msgopt).callback_function = "TestMessage_callback";

    optional string stringvalue = 1;
    repeated int32 int32value = 2;
    repeated fixed32 fixed32value = 3;
    repeated fixed64 fixed64value = 4;
    optional SubMessage submsg = 5;
    repeated string repeatedstring = 6;
}

